### PR TITLE
#405 Detail page - career position

### PR DESCRIPTION
**Pull request by jankaBehal**
Getting data for open positions detail page
Creating content for the page
Styling the page
Installing new packages
Current missing functionality
- Header image - waiting for communication
- Breadcrumbs - static only
- Responsive design - in next version
- Styling of some HTML elements - in next version

### CODE REVIEW

File: next/src/pages/career/[id].tsx [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+
+const Page = ({ entity, navigation, general }: PageProps) => {
+  const {
+    isError,
+    isPending,
+    error,
+    data: positionDetail,
+  } = useQuery({
+    queryKey: ['PositionsDetail', entity.id],
+    queryFn: () => fetchPositionsDetailFromApi(entity.id),
```
Comment (radoslavzeman) [inline next/src/pages/career/[id].tsx:100]: This fetch should happen actually server-side (in `getStaticProps`) and `positionDetails` could be returned similarly as `entity` is return in other detail pages.

File: next/src/pages/career/[id].tsx [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+  })
+
+  const { t } = useTranslation()
+  const hasData = !isError && !isPending
+  const HTML_REGEX = /(<([^>]+)>)/gi
+
+  const breadcrumbs = useMemo(
+    () => [
+      ...getPageBreadcrumbs('o-nas/kariera', navigation.pagePathsMap),
```
Comment (radoslavzeman) [inline next/src/pages/career/[id].tsx:109]: If it's only MVP phase, we always try to add a TODO comment at least, to know that this is not a desired solution.
(In final implementation, do not hardcode specific page urls)

File: next/src/pages/career/[id].tsx [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+  )
+
+  const title = isError ? error.message : isPending ? t('common.loading') : positionDetail.name
+
+  return (
+    <GeneralContextProvider general={general} navigation={navigation}>
+      {/* TODO common Head/Seo component */}
+      <Head>
+        <title>{t('career.detailHeadline')}</title>
+      </Head>
```
Comment (radoslavzeman) [inline next/src/pages/career/[id].tsx:122]: You can already use `SeoHead`

File: next/src/pages/career/[id].tsx [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+        <SectionContainer background="primary">
+          <div
+            id="career-section-content"
+            className="relative mx-auto max-w-screen-xl bg-background-primary px-0 py-6 lg:py-12"
+          >
+            <div className="flex flex-col items-start gap-4 md:flex-row lg:gap-8">
+              <div className="flex w-full shrink flex-col md:w-[50rem]">
+                {hasData && (
+                  <>
+                    <div className="flex flex-row flex-wrap justify-between rounded-lg border border-border-default">
```
Comment (radoslavzeman) [inline next/src/pages/career/[id].tsx:143]: Wouldn't `grid` be better for this case?

File: next/src/services/nalgoo/fetchPositionsDetail.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+/**
+ * Fetches and parses client data from the Nalgoo client.
+ * It's wrapped into function to use it both directly and in API handler.
+ *
+ */
+export const fetchPositionsDetail = async (id: string | undefined | string[]) => {
+  // TODO: this fetch should not directly fetch the data, but use OpenAPI generation and client with requests
+
+  const response = await fetch(
+    `${process.env.NEXT_NALGOO_URL}/job-offer/${id}?expand=regions,employment_forms,positions,min_education,educations,workMode,position_type,business_unit,custom_fields&api_key=${process.env.NEXT_PUBLIC_NALGOO_API_KEY}`,
```
Comment (radoslavzeman) [inline next/src/services/nalgoo/fetchPositionsDetail.ts:10]: Please add some comment where we can find more information about `expand` attribute

File: next/src/services/nalgoo/fetchOpenPositions.ts [hunk 6]
```diff
@@ -6,7 +6,7 @@
 export const fetchOpenPositions = async () => {
   // TODO: this fetch should not directly fetch the data, but use OpenAPI generation and client with requests
   const response = await fetch(
-    `${process.env.NEXT_NALGOO_URL}/job-offer?api_key=${process.env.NEXT_PUBLIC_NALGOO_API_KEY}`,
+    `${process.env.NEXT_NALGOO_URL}/job-offer?expand=company_sectors,regions,employment_forms,position_type,business_unit,custom_fields&api_key=${process.env.NEXT_PUBLIC_NALGOO_API_KEY}`,
```
Comment (radoslavzeman) [inline next/src/services/nalgoo/fetchOpenPositions.ts:9]: Please add some comment where we can find more information about expand attribute

File: next/src/assets/icons-olo/career-salary.svg
[the remark this answers is not in the document]
Comment (jankaBehal) [reply next/src/assets/icons-olo/career-salary.svg:1]: ok, icons are updated

File: next/src/components/common/Card/CareerRowCard.tsx [hunk 7]
```diff
@@ -15,25 +17,20 @@ type CareerRowCardProps = {
...
-          {/* TODO Change later to Tooltip component */}
-          {/* TODO dashed border */}
-          {toolTipText ? (
-            <div title={toolTipText}>
-              <Icon name="pomoc" className="size-5 lg:size-6" />
-            </div>
-          ) : null}
-        </div>
+    <CardBase variant="unstyled" className={cn('flex w-1/2 flex-row gap-4 p-6', className)}>
+      <div className="w-6">{icon}</div>
```
[the remark this answers is not in the document]
Comment (jankaBehal) [reply next/src/components/common/Card/CareerRowCard.tsx:23]: done

File: next/src/components/common/Card/CareerRowCard.tsx [hunk 8]
```diff
@@ -15,25 +17,20 @@ type CareerRowCardProps = {
...
-          </div>
-          {/* TODO Change later to Tooltip component */}
-          {/* TODO dashed border */}
-          {toolTipText ? (
-            <div title={toolTipText}>
-              <Icon name="pomoc" className="size-5 lg:size-6" />
-            </div>
-          ) : null}
-        </div>
+    <CardBase variant="unstyled" className={cn('flex w-1/2 flex-row gap-4 p-6', className)}>
```
[the remark this answers is not in the document]
Comment (jankaBehal) [reply next/src/components/common/Card/CareerRowCard.tsx:22]: I added grid for layout, but left flex and padding in the card

File: next/src/pages/career/[id].tsx (hunk 3 above)
Comment (jankaBehal) [reply next/src/pages/career/[id].tsx:122]: done

File: next/src/pages/career/[id].tsx (hunk 4 above)
Comment (jankaBehal) [reply next/src/pages/career/[id].tsx:143]: for the sidebar - content part, or for four info cards?

File: next/src/pages/career/[id].tsx (hunk 2 above)
Comment (jankaBehal) [reply next/src/pages/career/[id].tsx:109]: done

File: next/src/services/nalgoo/fetchOpenPositions.ts (hunk 6 above)
Comment (jankaBehal) [reply next/src/services/nalgoo/fetchOpenPositions.ts:9]: added

File: next/src/pages/career/[id].tsx (hunk 1 above)
Comment (jankaBehal) [reply next/src/pages/career/[id].tsx:100]: moved to the getStaticProps

File: next/src/components/common/Card/CareerRowCard.tsx [hunk 9]
```diff
@@ -19,8 +19,8 @@ type CareerRowCardProps = {
 
 const CareerRowCard = ({ value, label, toolTipText, icon, className }: CareerRowCardProps) => {
   return (
-    <CardBase variant="unstyled" className={cn('flex w-1/2 flex-row gap-4 p-6', className)}>
-      <div className="w-6">{icon}</div>
+    <CardBase variant="unstyled" className={cn('flex flex-row gap-4 p-6', className)}>
```
Comment (radoslavzeman) [inline next/src/components/common/Card/CareerRowCard.tsx:22]: `flex-row` is default, no need to specify it

File: next/src/pages/career/[id].tsx [hunk 10]
```diff
@@ -129,7 +113,7 @@ const Page = ({ entity, navigation, general }: PageProps) => {
       <PageLayout>
         <SectionContainer background="secondary">
           <Breadcrumbs breadcrumbs={breadcrumbs} />
-          <HeaderTitleText title={title ?? ''} />
+          <HeaderTitleText title={positionDetail?.name || ''} />
```
Comment (radoslavzeman) [inline next/src/pages/career/[id].tsx:116]: `??`

File: next/src/pages/career/[id].tsx [hunk 11]
```diff
@@ -0,0 +1,244 @@
...
+
+type StaticParams = {
+  id: string
+}
+
+export const getStaticPaths: GetStaticPaths<StaticParams> = async () => {
+  return { paths: [], fallback: 'blocking' }
+}
+
+export const getStaticProps: GetStaticProps<PageProps, StaticParams> = async ({
```
Comment (radoslavzeman) [inline next/src/pages/career/[id].tsx:47]: Add comment to use `getServersideProps` or use them directly.

**Review by radoslavzeman — APPROVED** [review]
Very nice!

File: next/src/pages/career/[id].tsx (hunk 11 above)
Comment (radoslavzeman) [reply next/src/pages/career/[id].tsx:47]: Or add a comment to `getStaticProps` to fetch all open positions.

**Pull request merged**